cms/toolbar/utils.py

```diff
@@ -117,14 +119,42 @@ def get_toolbar_from_request(request):
     return getattr(request, 'toolbar', EmptyToolbar(request))
 
 
+def get_querystring_modifier(obj, url, language=None):
+    """
+    Get the querystring modifier if one exists for the content type, and add it to the endpoint url
+    """
+    qsp_config = getattr(settings, 'CMS_ENDPOINT_QUERYSTRING_CONFIGURATION', None)
+    search_field = f"{obj._meta.app_label}.{obj._meta.model_name}"
+    if not qsp_config:
+        raise ImproperlyConfigured(
+            """
+            With 'CMS_ENDPOINT_QUERYSTRING_PARAM_ENABLED' set to True,
+            CMS_ENDPOINT_QUERYSTRING_CONFIGURATION must be provided
+            """
+        )
+    method = qsp_config.get(search_field, None)
+
+    if method:
+        param, content = method(obj, language)
+        urls = url.split('?')
+        if len(urls) > 1:
+            url += f'&{param}={content}'
+        else:
+            url += f'?{param}={content}'
+    return url
+
+
 def get_object_edit_url(obj, language=None):
     content_type = ContentType.objects.get_for_model(obj)
 
     if not language:
         language = get_language()
 
     with force_language(language):
-        return admin_reverse('cms_placeholder_render_object_edit', args=[content_type.pk, obj.pk])
+        url = admin_reverse('cms_placeholder_render_object_edit', args=[content_type.pk, obj.pk])
+    if getattr(settings, 'CMS_ENDPOINT_QUERYSTRING_PARAM_ENABLED', False):
+        url = get_querystring_modifier(obj, url, language)
```

Comment:
I may have missed some conversations about this, but is the intention for the solution to still allow a custom method to be passed in? Because as I understood, we only need to add the live url for a Page? In which case, the method injection could be removed, and instead we can handle getting the live url and appending it internally here? For example then your conditional could be more like:
```
    if getattr(settings, 'CMS_ADD_PAGE_LIVE_URL_PARAM', False) and isinstance(obj, Page):
        url = add_page_live_url_querystring_param(page=obj, url=url, language=language)

    return url
```
And then to use some of your existing changes but without the injection might be something like:
```
def add_page_live_url_querystring_param(page, url, language=None):
    live_url = page.get_absolute_url(language=language)
    # or could use a setting for the key?
    live_url_param = f"live-url={live_url}"
    urls = url.split('?')
    if len(urls) > 1:
        url += f'&{live_url_param}'
    else:
        url += f'?{live_url_param}'
    return url
```